### PR TITLE
Added date filtering option and date sorting option

### PR DESCRIPTION
**Pull request by rpeyron**

### CODE REVIEW

File: location_history_json_converter.py [hunk 1]
```diff
@@ -56,6 +73,11 @@ def main(argv):
             return
 
         items = data["locations"]
+        
+        if args.chronological:
+            if args.startdate or args.enddate:
+                items = [ item for item in items if dateCheck(item["timestampMs"]) ]
```
Comment (Scarygami) [inline location_history_json_converter.py]: I think the filtering for `startdate`, `enddate` should be moved out of the `args.chronological` branch, which would have the advantage that all the `if not dateCheck(item["timestampMs"])  :  continue` lines can be removed

File: location_history_json_converter.py [hunk 2]
```diff
@@ -67,6 +89,7 @@ def main(argv):
             first = True
 
             for item in items:
+                if not dateCheck(item["timestampMs"]) :  continue
```
Comment (Scarygami) [inline location_history_json_converter.py]: Could be removed as per comment above.

File: location_history_json_converter.py [hunk 3]
```diff
@@ -84,6 +107,7 @@ def main(argv):
         if args.format == "csv":
             f_out.write("Time,Location\n")
             for item in items:
+                if not dateCheck(item["timestampMs"]) :  continue
```
Comment (Scarygami) [inline location_history_json_converter.py]: Could be removed as per comment above.

File: location_history_json_converter.py [hunk 4]
```diff
@@ -126,6 +151,7 @@ def main(argv):
             f_out.write("  </metadata>\n")
             if args.format == "gpx":
                 for item in items:
+                    if not dateCheck(item["timestampMs"]) :  continue
```
Comment (Scarygami) [inline location_history_json_converter.py]: Could be removed as per comment above.

File: location_history_json_converter.py [hunk 5]
```diff
@@ -94,6 +118,7 @@ def main(argv):
             f_out.write("  <Document>\n")
             f_out.write("    <name>Location History</name>\n")
             for item in items:
+                if not dateCheck(item["timestampMs"]) :  continue
```
Comment (Scarygami) [inline location_history_json_converter.py]: Could be removed as per comment above.

File: location_history_json_converter.py [hunk 6]
```diff
@@ -149,8 +175,9 @@ def main(argv):
                 # The deltas below assume input is in reverse chronological order.  If it's not, uncomment this:
                 # items = sorted(data["data"]["items"], key=lambda x: x['timestampMs'], reverse=True)
                 for item in items:
+                    if not dateCheck(item["timestampMs"]) :  continue
```
Comment (Scarygami) [inline location_history_json_converter.py]: Could be removed as per comment above.

**Comment by askariikeka** [comment]
What is this ?

**Comment by Scarygami** [comment]
@askariikeka apparently you decided to "watch" this repository a while back, which is why you get notifications when something happens: https://github.com/Scarygami/location-history-json-converter/watchers

You can unwatch the repository to stop receiving those mails.

**Comment by rpeyron** [comment]
Hi, I made the changes you suggested (and some benchmarks that confirm these changes) ; I am not sure the PR has taken these changes : is it ok or should I make a new PR ?

**Comment by rpeyron** [comment]
I forgot to mention in the log ; I also changed : 
- fixed the use of argv in arg_parser.parse_args   (and this is useful to use Timeit to bench)
- modifed the comment of gpxtracks order with the abs modification and --chronological new option

**Comment by Scarygami** [comment]
Thanks!

**Pull request merged**